Tests/BackendIntegrationTests/OtherIntegrationTests.swift

```diff
@@ -248,6 +248,97 @@ class OtherIntegrationTests: BaseBackendIntegrationTests {
         expect(stubbedRequestCount).to(equal(1)) // Just the original request
     }
 
+    // MARK: - Virtual Currencies
+    func testGetVirtualCurrenciesWithBalancesOfZero() async throws {
+        let appUserIDWith0BalanceCurrencies = "integrationTestUserWithAllBalancesEqualTo0"
+        let purchases = try self.purchases
+
+        _ = try await purchases.logIn(appUserIDWith0BalanceCurrencies)
+
+        purchases.invalidateVirtualCurrenciesCache()
+        let virtualCurrencies = try await purchases.virtualCurrencies()
+
+        expect(virtualCurrencies.all.count).to(equal(2))
+
+        let testCurrency = try XCTUnwrap(virtualCurrencies["TEST"])
+        expect(testCurrency.balance).to(equal(0))
+        expect(testCurrency.code).to(equal("TEST"))
+        expect(testCurrency.name).to(equal("Test Currency"))
+        expect(testCurrency.serverDescription).to(equal("This is a test currency"))
+
+        let testCurrency2 = try XCTUnwrap(virtualCurrencies["TEST2"])
+        expect(testCurrency2.balance).to(equal(0))
+        expect(testCurrency2.code).to(equal("TEST2"))
+        expect(testCurrency2.name).to(equal("Test Currency 2"))
+
+        // swiftlint:disable:next todo
+        // TODO: We probably want to check that this is nil instead, depending on
+        // the results of a discussion with the backend team
+        expect(testCurrency2.serverDescription).to(beEmpty())
+    }
+
+    func testGetVirtualCurrenciesWithBalancesWithNonZeroValues() async throws {
+        let appUserIDWith0BalanceCurrencies = "integrationTestUserWithAllBalancesNonZero"
+        let purchases = try self.purchases
+
+        _ = try await purchases.logIn(appUserIDWith0BalanceCurrencies)
+
+        purchases.invalidateVirtualCurrenciesCache()
+        let virtualCurrencies = try await purchases.virtualCurrencies()
+
+        expect(virtualCurrencies.all.count).to(equal(2))
+
+        let testCurrency = try XCTUnwrap(virtualCurrencies["TEST"])
+        expect(testCurrency.balance).to(equal(100))
+        expect(testCurrency.code).to(equal("TEST"))
+        expect(testCurrency.name).to(equal("Test Currency"))
+        expect(testCurrency.serverDescription).to(equal("This is a test currency"))
+
+        let testCurrency2 = try XCTUnwrap(virtualCurrencies["TEST2"])
+        expect(testCurrency2.balance).to(equal(777))
+        expect(testCurrency2.code).to(equal("TEST2"))
+        expect(testCurrency2.name).to(equal("Test Currency 2"))
+
+        // swiftlint:disable:next todo
+        // TODO: We probably want to check that this is nil instead, depending on
+        // the results of a discussion with the backend team
+        expect(testCurrency2.serverDescription).to(beEmpty())
```

Comment:
We're going to change this to send down `null` from the backend. I'll update this to 

```suggestion
        expect(testCurrency2.serverDescription).to(beNil())
```

once that change has been deployed